File: go/check.go
package main

import (
  "fmt"
  "net/http"
  "encoding/json"
  "bytes"
  "io/ioutil"
)

func buildQuery(uploaderIds []string) string {
  query := fmt.Sprintf(`
    query {
      getF5498Statements(
        uploaderIds: [
          %s
        ]
      ) {
        errors
        statements {
          pageInfo {
            hasNextPage
          }
          nodes {
            otxId
            uploaderId
            statementValid
            validationMessages
            status
            statusDescription
						recipientFirstName
						recipientLastName
						recipientTin
						recipientTinType
						recipientAddress1
						recipientAddress2
						recipientCity
						recipientState
						recipientZipCode
						senderName
						senderTin
						senderAddress1
						senderCity
						senderState
						senderZipCode
						lateContributionCode
						lateContributionYear
						ira
						roth
          }
        }
      }
    }      
  `, formatIds(uploaderIds) )
  return(query)
}

func postGql(gql string, credential Credential) CheckResponse {
  endpoint := getGraphqlEndpoint()
  bodyReader := bytes.NewReader([]byte(formatForBody(gql)))

  req, _ := http.NewRequest(http.MethodPost, endpoint, bodyReader)
  req.Header.Set("Content-Type", "application/json")
  req.Header.Set("access-token", credential.accessToken)
  req.Header.Set("client", credential.client)
  req.Header.Set("uid", credential.uid)
  res, _ := http.DefaultClient.Do(req)
  body, _ := ioutil.ReadAll(res.Body)
  var result CheckResponse
  json.Unmarshal(body, &result)
  return(result)
}

func main() {
	credential := getCredential()
	// See the file ../data/f1098e-data.json for statement data and format.
	uploaderIds := []string{"6733766","6733767","6733768","6733769","6733710"}
	query := buildQuery(uploaderIds)
	response := postGql(query, credential)
  // response is an CheckResponse structure that you can manipulate
  // to suit your needs.  See structures.go for the format.
  // Here we're just going to print it (as a JSON object to make it
  // easier to read).
	formattedResponse, _ := json.MarshalIndent(response, "", "  ")
	fmt.Println(string(formattedResponse))
  // Credentials are valid for several days after they've been issued.
  // If you plan to do more work, you can use the same credential.  When done 
  // working, you should delete the credential to provide additional security.
  if del := deleteCredential(credential); del == true {
  	fmt.Println("\nSuccessfully deleted credential.")
  }
}

File: java/Check.java
import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

import com.ottertax.support.Credential;
import com.ottertax.support.GetF5498StatementsResponse;

public class Check {
  private static String buildQuery(String[] uploaderIds) {
    Gson gson = new Gson();
    StringBuilder sb = new StringBuilder();
    sb.append("query {\n" +
              "  getF5498Statements(\n" +
              "    uploaderIds: [\n" );
    for(String uploaderId : uploaderIds) {
      sb.append(gson.toJson(uploaderId) + " ");
    }
    sb.append("\n    ]\n" +
              "  ) {\n" +
              "    errors\n" +
              "    statements {\n" +
              "      pageInfo {\n" +
              "        hasNextPage\n" +
              "      }\n" +
              "      nodes {\n" +
              "        otxId\n" + 
              "        uploaderId\n" + 
              "        statementValid\n" + 
              "        validationMessages\n" + 
              "        status\n" + 
              "        statusDescription\n" + 
              "        recipientFirstName\n" + 
              "        recipientLastName\n" + 
              "        recipientTin\n" + 
              "        recipientTinType\n" + 
              "        recipientAddress1\n" + 
              "        recipientAddress2\n" + 
              "        recipientCity\n" + 
              "        recipientState\n" + 
              "        recipientZipCode\n" + 
              "        senderName\n" + 
              "        senderTin\n" + 
              "        senderAddress1\n" + 
              "        senderCity\n" + 
              "        senderState\n" + 
              "        senderZipCode\n" + 
              "        lateContributionCode\n" + 
              "        lateContributionYear\n" + 
              "        ira\n" + 
              "        roth\n" + 
              "      }\n" +
              "    }\n" +
              "  }\n" +
              "}\n");
    return(sb.toString());
  }

  public static void main(String[] args) {
    Helper helper = Helper.getInstance();
    Credential credential = helper.getCredential();
    String[] uploaderIds = {"6733766","6733767","6733768","6733769","6733710"};
    String query = buildQuery(uploaderIds);
    String response = helper.postGraphql(credential, query);

    Gson gson = new GsonBuilder().setPrettyPrinting().create();
    GetF5498StatementsResponse getF5498StatementsResponse = gson.fromJson(response, GetF5498StatementsResponse.class);
    System.out.println(gson.toJson(getF5498StatementsResponse));

    if(helper.deleteCredential(credential) == true) {
      System.out.println("\nSuccessfully deleted credential.");
    }
  }

}
